sycl/source/handler.cpp

```diff
@@ -635,29 +627,32 @@ event handler::finalize() {
 
       if (DiscardEvent) {
         EnqueueKernel();
-#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
-        LastEventImpl->setStateDiscarded();
```

Comment:
Line 654 on the right does `create_discarded_event` instead, which, IMO, is even more readable.